src/games/citations.c

```diff
@@ -383,14 +384,20 @@ int igraph_citing_cited_type_game(igraph_t *graph, igraph_integer_t nodes,
                                   igraph_bool_t directed) {
 
     igraph_vector_t edges;
-    igraph_i_citing_cited_type_game_struct_t str = { 0, 0 };
+    igraph_i_citing_cited_type_game_struct_t str = { 0, NULL };
     igraph_psumtree_t *sumtrees;
     igraph_vector_t sums;
     long int nocats;
     long int i, j;
 
+    if (nodes == 0) {
+        return igraph_empty(graph, 0, directed);
+    }
     if (igraph_vector_size(types) != nodes) {
-        IGRAPH_ERROR("Invalid size of types", IGRAPH_EINVAL);
+        IGRAPH_ERRORF("Length of types vector (%ld), not equal to number of nodes (%" IGRAPH_PRId ").", IGRAPH_EINVAL, igraph_vector_size(types), nodes);
```

Comment:
There's an unnecessary comma in this message after the first `(%ld)`.

Let's also break the line, it's very long.